src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -961,25 +961,79 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
     return matched
 
 
-def create_function(resource_group_name, new_app_name, storage_account, plan=None):
+def create_function(resource_group_name, name, storage_account, plan=None,
+                    consumption_plan_location=None):
+
+    if plan is not None and consumption_plan_location is not None:
+        raise CLIError("Use Plan or Consumption Plan Location")
+
+    webapp_def = Site(location='')
     client = web_client_factory()
-    if is_valid_resource_id(plan):
-        plan = parse_resource_id(plan)['name']
-    location = _get_location_from_app_service_plan(client, resource_group_name, plan)
-    webapp_def = Site(server_farm_id=plan, location=location)
+    if consumption_plan_location is not None:
+        locations = list_consumption_locations()
+        location = next((l for l in locations if l['name'] == consumption_plan_location), None)
+        if location is None:
+            raise CLIError("Location is invalid. Use: az functionapp list-consumption-locations")
+        webapp_def.location = consumption_plan_location
+    else:
+        if is_valid_resource_id(plan):
+            plan = parse_resource_id(plan)['name']
+        location = _get_location_from_app_service_plan(client, resource_group_name, plan)
+        webapp_def.server_farm_id = plan
+        webapp_def.location = location
```

Comment:
you need to error out when `--consumption_plan_location` and `--plan` are both missing